js/sts-hotspot.js

```diff
@@ -0,0 +1,72 @@
+AFRAME.registerComponent('sts-hotspot', {
+  schema: {
+    show: { type: 'selector' },
+    sound: { type: 'selector' },
+    playing: { default: false }
+  },
+
+  init: function () {
+    var self = this;
+    var popupEl = this.data.show;
+
+    popupEl.dataset.hotspot = self.el.id;
+
+    self.el.addEventListener('mouseenter', function () {
+      self.el.setAttribute('sts-hotspot', { playing: true });
+      self.el.dataset.playing = true;
+
+      popupEl.setAttribute('material', { visible: true });
+      popupEl.setAttribute('visible', true);
+    });
+
+    window.requestIdleCallback(this.preloadAsset.bind(this));
```

Comment:
to [support Edge](https://caniuse.com/#feat=requestidlecallback), can you do this conditional check?

```js
if (window.requestIdleCallback) {
  window.requestIdleCallback(this.preloadAsset.bind(this));
} else {
  this.preloadAsset();
}
```

p.s. :heart: for using `requestIdleCallback` to improve the perf. I look forward to seeing other devs start adopting the API too.